File: pl-domain/country-service/src/main/java/com/polylab/countries/common/configuration/MapperConfiguration.java
package com.polylab.countries.common.configuration;

import org.modelmapper.ModelMapper;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;

@Configuration
public class MapperConfiguration {

    @Bean
    @Primary
    public ModelMapper modelMapper() {
        return new ModelMapper();
    }


}

File: jx-core/src/main/kotlin/com/jxareas/jxcore/configuration/mapper/MapperConfiguration.kt
package com.jxareas.jxcore.configuration.mapper

import org.modelmapper.ModelMapper
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary


/**
 * Basic configuration class for Mappers.
 */
@Configuration
class MapperConfiguration {
    /**
     * Creates a simple `ModelMapper` bean.
     *
     * @return The `ModelMapper` bean instance.
     */
    @Primary
    @Bean
    fun modelMapper(): ModelMapper = ModelMapper()
}

